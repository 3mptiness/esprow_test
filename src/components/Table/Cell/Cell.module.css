label {
	display: flex;
}

td {
	border: 1px solid lightgray;
}

.RadioCell {
	display: flex;
	flex-direction: column;
	justify-content: space-evenly;
	align-items: end;
}

.Input {
	width: 100%;
	height: 100%;
	border: none;
}
