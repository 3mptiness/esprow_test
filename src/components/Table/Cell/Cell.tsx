import React, { ChangeEvent, FC, memo } from 'react';
import { ChangeHandler } from '../../../types';
import css from './Cell.module.css';

export const Cell: FC<{
	id: string;
	index: number;
	type: string;
	title: string;
	value: string | number;
	onChange?: ChangeHandler;
	height?: number;
}> = memo((props) => {
	const {
		id, index, title, type, value, onChange, height,
	} = props;

	const handleBlur = (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) =>
		onChange && onChange({ index, prop: title, value: e.target.value });

	if (type === 'textarea') {
		return (
			<td style={{ height }}>
				<textarea
					defaultValue={value}
					className={css.Input}
					name={id + title}
					onBlur={handleBlur}
				/>
			</td>
		);
	}

	if (type === 'radio') {
		return (
			<td style={{ height }}>
				<div className={css.RadioCell}>
					<label htmlFor={`${id + title}true`}>
						<span>true</span>
						<input
							className={css.Input}
							id={`${id + title}true`}
							name={id + title}
							type="radio"
							value="true"
							defaultChecked={`${value}` === 'true'}
							onBlur={handleBlur}
						/>
					</label>
					<label htmlFor={`${id + title}false`}>
						<span>false</span>
						<input
							className={css.Input}
							id={`${id + title}false`}
							name={id + title}
							type="radio"
							value="false"
							defaultChecked={`${value}` === 'false'}
							onBlur={handleBlur}
						/>
					</label>
				</div>
			</td>
		);
	}

	return (
		<td style={{ height }}>
			<input
				className={css.Input}
				name={id + title}
				type={type}
				defaultValue={value}
				onBlur={handleBlur}
			/>
		</td>
	);
});
