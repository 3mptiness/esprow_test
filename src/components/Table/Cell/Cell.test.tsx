import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { Cell } from './Cell';

const cellProps = {
	id: '100500', index: 10, title: 'title', height: 20,
};

describe('Cell', () => {
	it('render input', () => {
		expect(
			createRenderer().render(
				<Cell {...cellProps} type="input" value="value" />
			)
		).toMatchSnapshot();
	});

	it('render radio', () => {
		expect(
			createRenderer().render(<Cell {...cellProps} type="radio" value="true" />)
		).toMatchSnapshot();
	});

	it('render textarea', () => {
		expect(
			createRenderer().render(
				<Cell {...cellProps} type="textarea" value="long text" />
			)
		).toMatchSnapshot();
	});
});
