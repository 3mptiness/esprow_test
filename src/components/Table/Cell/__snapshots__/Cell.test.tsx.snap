// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell render input 1`] = `
<td
  style={
    Object {
      "height": 20,
    }
  }
>
  <input
    className="Input"
    defaultValue="value"
    name="100500title"
    onBlur={[Function]}
    type="input"
  />
</td>
`;

exports[`Cell render radio 1`] = `
<td
  style={
    Object {
      "height": 20,
    }
  }
>
  <div
    className="RadioCell"
  >
    <label
      htmlFor="100500titletrue"
    >
      <span>
        true
      </span>
      <input
        className="Input"
        defaultChecked={true}
        id="100500titletrue"
        name="100500title"
        onBlur={[Function]}
        type="radio"
        value="true"
      />
    </label>
    <label
      htmlFor="100500titlefalse"
    >
      <span>
        false
      </span>
      <input
        className="Input"
        defaultChecked={false}
        id="100500titlefalse"
        name="100500title"
        onBlur={[Function]}
        type="radio"
        value="false"
      />
    </label>
  </div>
</td>
`;

exports[`Cell render textarea 1`] = `
<td
  style={
    Object {
      "height": 20,
    }
  }
>
  <textarea
    className="Input"
    defaultValue="long text"
    name="100500title"
    onBlur={[Function]}
  />
</td>
`;
